products/memgraph/v0.51.0/how_to_guides/use-and-implement-query-modules.md

```diff
@@ -0,0 +1,322 @@
+Assuming the standard installation on Debian, you would run Memgraph with the
+following command.
+
+```plaintext
+systemctl start memgraph
+```
+
+When using Docker, the equivalent would be the following.
+
+```plaintext
+docker run -p 7687:7687 \
+  -v mg_lib:/var/lib/memgraph -v mg_log:/var/log/memgraph -v mg_etc:/etc/memgraph \
+  memgraph
+```
+
+Memgraph will now attempt to load the query modules form all `*.so` and `*.py`
+files it finds in the default (`/usr/lib/memgraph/query_modules`) directory.
+The `*.so` modules are written using the C API and the `*.py` modules are
+written using the Python API. Each file corresponds to one query module. Names
+of these files will be mapped to query module names.  So in our case, we have
+an `example.so` which will be mapped to `example` module and a `py_example.py`
+which will be mapped to `py_example` module in the query language.
+
+Each query module can define multiple procedures. Both of our examples define
+a single procedure creatively named `procedure`.
+
+If you want to change the directory in which Memgraph searches for query
+modules, just change the `--query-modules-directory` flag in the main
+configuration file (`/etc/memgraph/memgraph.conf`) or supply it as
+a command-line parameter (e.g. when using Docker).
+
+### Syntax for Calling Procedures
+
+OpenCypher has a special syntax for calling procedures in loaded query
+modules. For example:
+
+```opencypher
+CALL example.procedure("string-argument") YIELD args, result;
+```
+
+Each procedure returns zero or more records, where each record contains named
+fields. The `YIELD` part is used to select fields we are interested in. If we
+are not interested in any fields, the `YIELD` part can be omitted. The
+procedure will still run in such a case, but the record fields will not be
+stored in variables. In the above example, we assume that `example.procedure`
+will produce a record with 2 fields, `args` and `result`. Trying to `YIELD`
+fields which are not part of the produced record will result in an error.
+
+Procedures may be called standalone as in the above example, or as a part of a
+larger query. This is useful if we want the procedure to work on data the
+query is producing. For example:
+
+```
+MATCH (node) CALL example.procedure(node) YIELD result RETURN *;
+```
+
+Unfortunately, when we use `CALL` in a larger query, we have to explicitly
+`RETURN` from the query to get the results. Naturally, the `RETURN` is not
+needed if we perform updates after `CALL`. This follows the openCypher
+convention that read-only queries need to end with a `RETURN`, while queries
+which update something don't need to `RETURN` anything.
+
+If a procedure returns a record with a field name that may clash with some
+variable we already have in a query, that field name can be aliased into some
+other name. For example:
+
+```opencypher
+MATCH (node) CALL example.procedure(42) YIELD node AS result RETURN *;
```

Comment:
I would rewrite this query so that it is a valid query when executed with the built-in `example.procedure` procedure.
`MATCH (result) CALL example.procedure(42) YIELD result AS procedure_result RETURN *;`